src/items/qnitegrid.cpp

```diff
@@ -19,3 +21,15 @@ QNanoQuickItemPainter* QniteGrid::createItemPainter() const
 {
     return new QniteGridPainter;
 }
+
+void QniteGrid::setDrawXAxes(bool value)
+{
+    m_drawXAxes = value;
+    emit drawXAxesChanged();
```

Comment:
Check if the new value if different before assigning and emitting signal, it would be useless to emit a signal if the value doesn't change.